### PR TITLE
SOLR-16318: WIP - Use up to JDK 11 features

### PR DESCRIPTION
**Pull request by risdenk**
https://issues.apache.org/jira/browse/SOLR-16318

# Description

* Use enhanced for loops
* Avoid type declaration when `<>` can be used
* Use `computeIfAbsent`
* use `Objects.requireNonNullElse` to simplify if/else statements
* Simplify lambda statements
* `Collections.sort` -> `list.sort`
* Use string `.contains` instead of checking `indexOf`
* Avoid unnecessary boxing/unboxing

### CODE REVIEW

**Review by madrob — COMMENTED** [review]
Progress: 100 files viewed. Some comments on specific lines.

File: solr/core/src/java/org/apache/solr/filestore/PackageStoreAPI.java [hunk 1]
```diff
@@ -212,12 +212,10 @@ public void upload(SolrQueryRequest req, SolrQueryResponse rsp) {
         } catch (IOException e) {
           throw new SolrException(SolrException.ErrorCode.BAD_REQUEST, e);
         }
-      } catch (InterruptedException e) {
-        log.error("Unexpected error", e);
       } catch (KeeperException.NodeExistsException e) {
         throw new SolrException(
             SolrException.ErrorCode.SERVER_ERROR, "A write is already in process , try later");
-      } catch (KeeperException e) {
+      } catch (InterruptedException | KeeperException e) {
```
[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/java/org/apache/solr/filestore/PackageStoreAPI.java:218]: Thanks yea I noticed this as well. You are quick to review  :) - I threw it up so I could more easily look at the changes too.

**Comment by risdenk** [comment]
Thanks @madrob for the quick review. I threw it up to review myself and wasn't expecting comments yet :) I agree with your findings and many probably repeat throughout. I'll clean it up over the next few days.

**Comment by epugh** [comment]
I am not really up on the new stuff in JDK 11 @risdenk so probably not the best person to review... ;-)

**Comment by risdenk** [comment]
@epugh mostly an FYI since you were going through intelli warnings - these are also intellij warnings/findings behind the scenes.

**Pull request closed without merging**